components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/PublisherCommonUtils.java

```diff
@@ -850,6 +861,49 @@ public static boolean isValidWSAPI(APIDTO api) {
         return isValid;
     }
 
+    /**
+     * Validate sandbox and production endpoint URLs.
+     *
+     * @param apiDto API DTO of the API
+     * @return validity of URLs found within the endpoint configurations of the DTO
+     */
+    public static boolean validateEndpoints(APIDTO apiDto) {
+
+        ArrayList<String> endpoints = new ArrayList<>();
+        UrlValidator urlValidator = new UrlValidator(UrlValidator.ALLOW_ALL_SCHEMES +
+                UrlValidator.ALLOW_LOCAL_URLS);
```

Comment:
Is this "+" sign used to add multiple schemes?

Reply:
No, it is used to add `ALLOW_ALL_SCHEMES` and `ALLOW_LOCAL_URLS` options to the validator (official doc: https://commons.apache.org/proper/commons-validator/apidocs/org/apache/commons/validator/routines/UrlValidator.html). This constructor takes in a long value as the options. Without adding these 2 options, it will not accept urls from schemes like tcp, ws, wss, etc. (hence, ALLOW_ALL_SCHEMES) and localhost urls (hence, ALLOW_LOCAL_URLS).